From 807852134a37e7c24c62f126e2301368d0167f2c Mon Sep 17 00:00:00 2001
From: Kevin Wang <kevin1.wang@amd.com>
Date: Wed, 8 May 2019 14:37:08 +0800
Subject: [PATCH 2368/2940] drm/amd/powerplay: fix dpm freq unit error (10KHz
 -> Mhz)

the interface smu_v11_0_get_current_clk_freq should be return 10Khz not
Mhz unit to adapt vega20 and navi10 asic at the same time.

Signed-off-by: Kevin Wang <kevin1.wang@amd.com>
Acked-by: Huang Rui <ray.huang@amd.com>
Signed-off-by: Alex Deucher <alexander.deucher@amd.com>
---
 drivers/gpu/drm/amd/powerplay/navi10_ppt.c |  2 ++
 drivers/gpu/drm/amd/powerplay/smu_v11_0.c  | 19 ++++++++++---------
 2 files changed, 12 insertions(+), 9 deletions(-)

diff --git a/drivers/gpu/drm/amd/powerplay/navi10_ppt.c b/drivers/gpu/drm/amd/powerplay/navi10_ppt.c
index cea5704e3f4f..ddd683f3ceba 100644
--- a/drivers/gpu/drm/amd/powerplay/navi10_ppt.c
+++ b/drivers/gpu/drm/amd/powerplay/navi10_ppt.c
@@ -581,6 +581,8 @@ static int navi10_print_clk_levels(struct smu_context *smu,
 		ret = smu_get_current_clk_freq(smu, clk_type, &cur_value);
 		if (ret)
 			return size;
+		/* 10KHz -> MHz */
+		cur_value = cur_value / 100;
 
 		size += sprintf(buf, "current clk: %uMhz\n", cur_value);
 
diff --git a/drivers/gpu/drm/amd/powerplay/smu_v11_0.c b/drivers/gpu/drm/amd/powerplay/smu_v11_0.c
index 957fb63b8d5e..8203ebe52227 100644
--- a/drivers/gpu/drm/amd/powerplay/smu_v11_0.c
+++ b/drivers/gpu/drm/amd/powerplay/smu_v11_0.c
@@ -1098,16 +1098,17 @@ static int smu_v11_0_get_current_clk_freq(struct smu_context *smu,
 
 	/* if don't has GetDpmClockFreq Message, try get current clock by SmuMetrics_t */
 	if (smu_msg_get_index(smu, SMU_MSG_GetDpmClockFreq) == 0)
-		return smu_get_current_clk_freq_by_table(smu, clk_id, value);
-
-	ret = smu_send_smc_msg_with_param(smu, SMU_MSG_GetDpmClockFreq,
-					  (smu_clk_get_index(smu, clk_id) << 16));
-	if (ret)
-		return ret;
+		ret =  smu_get_current_clk_freq_by_table(smu, clk_id, &freq);
+	else {
+		ret = smu_send_smc_msg_with_param(smu, SMU_MSG_GetDpmClockFreq,
+						  (smu_clk_get_index(smu, clk_id) << 16));
+		if (ret)
+			return ret;
 
-	ret = smu_read_smc_arg(smu, &freq);
-	if (ret)
-		return ret;
+		ret = smu_read_smc_arg(smu, &freq);
+		if (ret)
+			return ret;
+	}
 
 	freq *= 100;
 	*value = freq;
-- 
2.17.1

